### PR TITLE
GPlus Badge and Twitter Timeline Widgets: Do not show widget title if the u…

### LINKED ISSUE #5753: GPlus and Twitter Widgets are showing before_title + title + after_title even if title is empty
GPlus and Twitter Widgets are showing before_widget + title + after_widget even if title is empty. This can cause little weird style issues.

Steps to reproduce: 
1. Add Twitter Timeline /Google Plus widget
2. Set an empty title for them
3. Empty markup is displayed above the widget but the content is empty

For instance, let's imagine this sidebar:
```
register_sidebar( array(
        ...
	'before_title'  => '<h2 class="widget-title">',
	'after_title'   => '</h2>',
        ...
) );
```

And this CSS:
```
h2.widget-title {
   padding:5px;
   background: #DDD;
}
```

If the widget title is empty, an empty box with background #DDD will appear above the widget.

I've noticed that all the rest of widgets are not showing the title if it's empty so for the sake of coherence a pull request is on the way.

Thanks.

### PR DESCRIPTION
**Pull request by igmoweb**
Fixes #5753

#### Changes proposed in this Pull Request:
Do not show `$args['before_title'] . $title . $args['after_title']` if the title is empty in the following widgets:
- Google+ Badge
- Twitter Timeline

#### Testing instructions:
Details here: [https://github.com/Automattic/jetpack/issues/5753](https://github.com/Automattic/jetpack/issues/5753)

#### Proposed changelog entry for your changes:
Widgets: Removed widget title if this is empty in Google Plus Badge and Twitter Timeline

### CODE REVIEW

File: modules/widgets/twitter-timeline.php [hunk 1]
```diff
@@ -80,9 +80,11 @@ public function widget( $args, $instance ) {
 
 		echo $args['before_widget'];
 
-		if ( isset( $instance['title'] ) ) {
+		$title = isset( $instance['title'] ) ? $instance['title'] : '';
+		$title = apply_filters( 'widget_title', $title );
+		if ( ! empty( $title ) ) {
 			/** This filter is documented in core/src/wp-includes/default-widgets.php */
```
Comment (jeherve) [inline modules/widgets/twitter-timeline.php]: Could you move the docblock so it remains on the line above the filter?

Thanks!

**Comment by igmoweb** [comment]
Whoops, done :)

**Comment by dereksmart** [comment]
I do agree that this should not show if empty, though removing it may alter the appearance of sites that already have set these widgets up and maybe added custom css around this that will shift things around.  

However, after speaking with @jeherve we are okay with making that move, as this is the correct functionality that is expected from widgets, and it is consistent with how our other widgets work.

**Comment by igmoweb** [comment]
Awesome. Thanks!

**Pull request merged**